sentinel-ai-models-simple-openai/src/test/java/com/phonepe/sentinelai/models/SimpleOpenAIModelStreamingTest.java

```diff
@@ -172,6 +146,82 @@ void testAgent(final WireMockRuntimeInfo wiremock) {
         log.info("Session stats: {}", stats);
     }
 
+    @NotNull
+    private static TestAgent setupAgent(final WireMockRuntimeInfo wiremock, final JsonMapper objectMapper, final OkHttpClient httpClient, final ExecutorService executor) {
+        final var model = new SimpleOpenAIModel<>(
+                "gpt-4o",
+                SimpleOpenAIAzure.builder()
+//                        .baseUrl("http://localhost:8080") //Wiremock recorder input: $AZURE_ENDPOINT
+//                        // Uncomment the above to record responses using the wiremock recorder.
+//                        // Yeah ... life is hard
+//                        .baseUrl(EnvLoader.readEnv("AZURE_ENDPOINT"))
+//                        .apiKey(EnvLoader.readEnv("AZURE_API_KEY"))
+                        .baseUrl(wiremock.getHttpBaseUrl())
+                        .apiKey("BLAH")
+                        .apiVersion("2024-10-21")
+                        .objectMapper(objectMapper)
+                        .clientAdapter(new OkHttpClientAdapter(httpClient))
+                        .build(),
+                objectMapper,
+                SimpleOpenAIModelOptions.builder()
+                        .toolChoice(SimpleOpenAIModelOptions.ToolChoice.AUTO)
+                        .build()
+        );
+        return new TestAgent(AgentSetup.builder()
+                                                .model(model)
+                                                .mapper(objectMapper)
+                                                .modelSettings(ModelSettings.builder()
+                                                                       .parallelToolCalls(false)
+                                                                       .temperature(0.1f)
+                                                                       .seed(1)
+                                                                       .build())
+                                                .executorService(executor)
+                                                .outputGenerationMode(OutputGenerationMode.STRUCTURED_OUTPUT)
+                                                .build());
+    }
+
+    @Test
+    @SneakyThrows
+    void testTimeouts(final WireMockRuntimeInfo wiremock) {
+        TestUtils.setupMocksWithFault(Fault.CONNECTION_RESET_BY_PEER);
+        final var response = execute(wiremock);
+        assertSame(ErrorType.TIMEOUT,
```

Comment:
Yeah .. as i mentioned above, this is not correct. If needed, we should rename timeout to COMMUNICATION_ERROR and club all types of network failures inside that.